OrbitCore/OrbitModule.cpp

```diff
@@ -134,58 +135,38 @@ std::string FindSymbols(const std::string& module_path) {
   return module_path;
 }
 
-//-----------------------------------------------------------------------------
-bool Pdb::LoadPdb(const char* a_PdbName) {
-  m_FileName = FindSymbols(a_PdbName);
-  m_Name = Path::GetFileName(m_FileName);
-  m_LoadedModuleName = a_PdbName;
+bool Pdb::LoadFunctions(const std::string& file_name) {
+  m_FileName = file_name;
+  m_Name = Path::GetFileName(file_name);
+  m_LoadedModuleName = file_name;
 
-  {
-    SCOPE_TIMER_LOG("nm");
-    // nm
-    // TODO: If we need linenumber information at some point, we need to find an
-    // alternative, as "nm -l" is super slow.
-    std::string nmCommand =
-        std::string("nm ") + m_FileName + std::string(" -n");
-    std::string nmResult = LinuxUtils::ExecuteCommand(nmCommand.c_str());
-    std::stringstream nmStream(nmResult);
-    std::string line;
-    while (std::getline(nmStream, line, '\n')) {
-      std::vector<std::string> tokens = Tokenize(line, " \t");
-      if (tokens.size() > 2)  // nm
-      {
-        const auto& name = tokens[2];
-        Function func(name, Path::GetFileName(a_PdbName),
-                      std::stoull(tokens[0], nullptr, 16), 0, this);
-        func.SetPrettyName(LinuxUtils::Demangle(name.c_str()));
-        this->AddFunction(func);
-      }
+  auto elf_file = ElfFile::Create(m_FileName);
+  if (elf_file == nullptr || !elf_file->GetFunctions(this, &m_Functions)) {
+    PRINT(absl::StrFormat("Unable to load functions from \"%s\"", m_FileName));
+    return false;
+  }
+
+  // For uprobes we need a function to be in the .text segment (why?)
+  // TODO: Shouldn't list of functions above be limited to .text segment?
```

Comment:
Meaning that we should make it like that or that it's supposed to be like that?

Reply:
I will clarify comment - what I mean here that maybe we should limit m_Functions with .text segment only, and remove IsAddressInTextSection here.